provider/azure/internal/azureauth/discovery.go

```diff
@@ -29,7 +29,9 @@ type recordAuthHeaderPolicy struct {
 
 func (p *recordAuthHeaderPolicy) Do(req *policy.Request) (*http.Response, error) {
 	resp, err := req.Next()
-	p.authHeader = resp.Header.Get(authenticateHeaderKey)
+	if resp.Header != nil {
```

Comment:
Um, this is bad - why are we ignoring the err here? This will panic is resp is nil

Reply:
Yep. If a http.Header is nil, Get is safe.

Reply:
Oops, guess it should be `resp != nil` instead? Looks like I assumed the wrong things is nil